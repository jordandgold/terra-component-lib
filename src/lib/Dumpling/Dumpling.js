import React from "react";
import PropTypes from "prop-types";
import "./Dumpling.scss";
import Icon from "../Icon/Icon";

const Dumpling = ({ title, link, size, icon, image }) => {
  return (
    <article className={`ter-dumpling ter-dumpling--${size}`}>
      {icon && (
        <div className="ter-dumpling__icon-wrapper">
          <Icon type={icon.type} size={icon.size} />
        </div>
      )}
      {image && (
        <img
          src={image.url}
          alt={image.altText}
          className="ter-dumpling__image"
        />
      )}
      {size === "small" && <div className={"ter-dumpling__small-dash"} />}
      <a
        className={`ter-dumpling__link ter-dumpling__link--${size}`}
        href={link.url}
      >
        {title}
      </a>
      <div />
    </article>
  );
};

export default Dumpling;

Dumpling.propTypes = {
  title: PropTypes.string.isRequired,
  link: PropTypes.shape({ url: PropTypes.string.isRequired }).isRequired,
  size: PropTypes.string.isRequired,
  icon: PropTypes.shape({
    type: PropTypes.string.isRequired,
    size: PropTypes.string.isRequired
  }),
  image: PropTypes.shape({
    url: PropTypes.string.isRequired,
    altText: PropTypes.string.isRequired
  })
};
