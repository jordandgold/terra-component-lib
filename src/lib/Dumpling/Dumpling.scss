@import "../global-styles/terra.scss";

.ter-dumpling {
  width: 200px;
  margin: auto;
  padding: 32px 0;

  &__icon-wrapper {
    padding-bottom: 24px;
    .ter-icon {
      margin: auto;
      display: block;
      fill: color("ui-1");
    }
  }

  &__small-dash {
    width: 25px;
    margin: auto;
    border-top: solid 1px color("ui-1");
  }

  &__link {
    padding-top: 16px;
    text-align: center;
    display: block;
    font-weight: font-weight("bold");
  }

  &__image {
    max-width: 75px;
    max-height: 75px;
    display: block;
    margin: 0 auto 32px;
  }
}
