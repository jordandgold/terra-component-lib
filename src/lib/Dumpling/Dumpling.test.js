import React from "react";
import Dumpling from "./Dumpling";
import { shallow } from "enzyme";
import { isIterable } from "core-js";

describe("Dumpling", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Dumpling
        icon={{ type: "enclosed-check-dark-32px", size: "32px" }}
        size="small"
        title="Dumplin'"
        link={{ url: "#" }}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot if it has an image", () => {
    wrapper = shallow(
      <Dumpling
        image={{ url: "www.test.com", altText: "I am alt text" }}
        size="small"
        title="Dumplin'"
        link={{ url: "#" }}
      />
    );
  });
});
