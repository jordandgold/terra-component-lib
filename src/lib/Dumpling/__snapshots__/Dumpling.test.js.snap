// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dumpling should match the snapshot 1`] = `
<article
  className="ter-dumpling ter-dumpling--small"
>
  <div
    className="ter-dumpling__icon-wrapper"
  >
    <Icon
      size="32px"
      type="enclosed-check-dark-32px"
    />
  </div>
  <div
    className="ter-dumpling__small-dash"
  />
  <a
    className="ter-dumpling__link ter-dumpling__link--small"
    href="#"
  >
    Dumplin'
  </a>
  <div />
</article>
`;
